File: java-prov/src/main/java/org/example/InvalidRecords.java
package org.example;

import org.openprovenance.prov.interop.InteropFramework;
import org.openprovenance.prov.model.Document;
import org.openprovenance.prov.model.Namespace;
import org.openprovenance.prov.model.ProvFactory;


public class InvalidRecords implements TestCase {

    public void serialize(String format) {
        //TODO
    }

    public void deserialize(String format) {
        var inf = new InteropFramework();
        Document document = inf.readDocumentFromFile(String.format("data/invalid_records.%s", format));
    }
}

File: python-prov/invalid_records.py
import datetime

import prov.dot
from prov.model import ProvDocument
import prov.graph

def perform():
    document = ProvDocument()
    ns = document.add_namespace("ex", "https://example.org/")
    a = document.activity("ex:a")
    document.wasGeneratedBy(a,a,datetime.datetime.now())
    document.serialize(r"..\java-prov\invalid_records.provn", format="provn")
    print(document.get_provn())